pkg/spec/config_linux.go

```diff
@@ -118,8 +119,41 @@ func (c *CreateConfig) addPrivilegedDevices(g *generate.Generator) error {
 		return err
 	}
 	g.ClearLinuxDevices()
-	for _, d := range hostDevices {
-		g.AddDevice(Device(d))
+
+	if rootless.IsRootless() {
+		mounts := make(map[string]interface{})
+		for _, m := range g.Mounts() {
+			mounts[m.Destination] = true
+		}
+		newMounts := []spec.Mount{}
+		for _, d := range hostDevices {
+			devMnt := spec.Mount{
+				Destination: d.Path,
+				Type:        TypeBind,
+				Source:      d.Path,
+				Options:     []string{"slave", "nosuid", "noexec", "rw", "rbind"},
+			}
+			if d.Path == "/dev/ptmx" || strings.HasPrefix(d.Path, "/dev/tty") {
+				continue
+			}
+			if _, found := mounts[d.Path]; found {
+				continue
+			}
+			st, err := os.Stat(d.Path)
+			if err != nil {
+				return errors.Wrapf(err, "stat %s", d.Path)
```

Comment:
Could the stat fail on EPERM? Should we just skip if so?

Reply:
I thought in that case we still want to skip the test.  It is more about checking if `/dev/kvm` is usable, that if it exists

Reply:
This isn't the test - this is the code for adding devices?

Reply:
yes sorry, I've added the code to skip EPERM.  It should not really happen unless `/dev` is not accessible and in that case we wouldn't be able even to use standard devices like `random` or `null`